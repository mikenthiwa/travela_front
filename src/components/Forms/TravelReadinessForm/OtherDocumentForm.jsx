import React from 'react';
import withTravelReadinessForm from '../../Hoc/withTravelReadinessForm';
import OtherDocumentFieldSet from './FormFieldsets/OtherDocumentFieldSet';

export default withTravelReadinessForm(OtherDocumentFieldSet,'other',
  {
    values: {
      name: '',
      dateOfIssue: '',
      expiryDate: '',
      documentId: ''
    },
    image: '',
    imageChanged: false,
    documentUploaded: false,
    uploadingDocument: false,
    errors: {},
    hasBlankFields: true,
    optionalFields: [
      'documentId',
      'dateOfIssue'
    ]
  });


