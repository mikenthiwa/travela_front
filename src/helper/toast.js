import toast from 'toastr';
import cancelBtn from '../images/toast-cancel.svg';

toast.options = {
  progressBar: false,
  closeButton: true,
  preventDuplicates: true,
  positionClass: 'toast-top-center',
  closeHtml: `
    <button class="custom-close-btn">
      <img alt="close btn"/>
    </button>
  `,
};

export const successMessage = message =>  toast.success(message);
export const errorMessage = message =>  toast.error(message);


