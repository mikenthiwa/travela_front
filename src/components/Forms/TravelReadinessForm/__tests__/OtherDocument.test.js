import React from 'react';
import toast from 'toastr';
import moxios from 'moxios';
import OtherDocumentForm from '../OtherDocumentForm';


const props = {
  errors: {},
  createTravelReadinessDocument: jest.fn(),
  closeModal: jest.fn(),
  openModal: jest.fn(),
  fetchUserData: jest.fn(),
  scanPassport: jest.fn(),
  user: {}, 
  showPassportForm:false,
  retrieving:false, 
  modalType:'add other',
  passportInfo:{
    passportData:{}
  }
};

const textFile = new Blob(['This is a text file'], {type : 'text/plain'});
textFile.name = 'textFile.txt';

const validFile = new Blob(['This is a valid png file'], {type : 'image/png', size: 1092});
validFile.name = 'file.png';

toast.error = jest.fn();
const event = {
  preventDefault: jest.fn(),
  target:{
    files: [validFile]
  }
};

describe('<OtherDocumentForm />', () => {
  const wrapper = mount(<OtherDocumentForm {...props} />);
  process.env.REACT_APP_CLOUNDINARY_API = 'https://api.cloudinary.com/v1_1/skybound/image/upload';
  beforeEach(() => {
    moxios.install();
  });

  afterEach(() => {
    moxios.uninstall();
    toast.error.mockReset();
  });
  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders file selector', () => {
    wrapper.find('.travel-document-select-file');
    expect(wrapper.find('.travel-document-select-file').length).toEqual(1);
  });


  it('closes modal on cancel', () => {
    expect(wrapper.find('#cancel').length).toEqual(1);
    wrapper.find('#cancel').simulate('click', event);
    expect(props.closeModal).toHaveBeenCalled();
  });

  it('handles submit', () => {
    moxios.stubRequest(
      process.env.REACT_APP_CLOUNDINARY_API,
      { status: 200, data: { url: 'url' } }
    );
    wrapper.setProps({document: {data: { imageName: 'image.jpg' }}});
    wrapper.state().file = validFile;
    wrapper.find('.travel-document-form').simulate('submit', event);
    moxios.wait(() => {
      expect(props.createTravelReadinessDocument).toHaveBeenCalled();
    });
  });

  it('changes state accordingly',  () => {
    moxios.stubRequest(
      process.env.REACT_APP_CLOUNDINARY_API,
      { status: 200, data: { url: 'url' } }
    );
    wrapper.find('#select-file').simulate('change', event);
    moxios.wait(() => {
      expect(wrapper.state().cloudinaryUrl).toEqual();
      expect(wrapper.find('.input-group').length).toEqual(1);
    });
  });

  it('toasts an error if file is invalid', () => {
    event.target.files = [textFile];
    wrapper.find('#select-file').simulate('change', event);
    expect(toast.error).toHaveBeenCalled();
  });

  it('toasts an error if cloudinary returns an error', () => {
    moxios.stubRequest(process.env.REACT_APP_CLOUNDINARY_API, { status: 500});
    wrapper.find('#select-file').simulate('change', event);
  });

  it('closes modal on cancel', () => {
    expect(wrapper.find('#cancel').length).toEqual(1);
    wrapper.find('#cancel').simulate('click', event);
    setTimeout(()=>{
      expect(props.closeModal).toHaveBeenCalled();
    }, 5000)
  });
});
