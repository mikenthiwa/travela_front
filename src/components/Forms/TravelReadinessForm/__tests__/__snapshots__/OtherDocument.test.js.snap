// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OtherDocumentForm /> renders without crashing 1`] = `
<div>
  <form
    className="travel-document-form"
    key=".0"
    onSubmit={[Function]}
  >
    <div
      className="travel-document-select-file"
    >
      <p
        className="attach-document-text"
      >
        Attach the image or PDF of your  document
      </p>
      <div
        className="document-input__input-container"
      >
        <label
          className="document-input__input-container__prompts"
          htmlFor="select-file"
        >
          <div
            className="document-input__input-container__prompts__img"
          >
            <img
              alt="Document Upload"
              src="document-upload-blue.svg"
            />
          </div>
          <div
            className="document-input__input-container__prompts__text"
          >
            <p>
              Choose from computer
            </p>
            <p
              className="maximum-file-size"
            >
              Maximum file size - 10MB
            </p>
          </div>
          <input
            accept="image/jpg, image/png, application/pdf"
            id="select-file"
            onChange={[Function]}
            type="file"
          />
        </label>
      </div>
    </div>
    <hr />
    <div
      className="travel-document-submit-area"
    >
      <fieldset>
        <div
          className="submit-area"
        >
          <button
            className="bg-btn bg-btn--inactive"
            disabled={false}
            id="cancel"
            onClick={[Function]}
            type="button"
          >
            Cancel
          </button>
          <button
            className="bg-btn bg-btn--active"
            disabled={true}
            id="submit"
            type="submit"
          >
            <undefined />
          </button>
        </div>
      </fieldset>
    </div>
  </form>
</div>
`;
